fix(mailbox): render inline images without Content-Disposition

nooty opened the pull request:
## Description
This pull request addresses issue #11948, where inline images embedded in emails (such as those sent from Outlook) are not rendered correctly if the Content-Disposition header is missing.

The solution ensures that images referenced via cid: in the HTML body are correctly identified and rewritten using url_for.

Fixes #11948

## Type of change
Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
Added test: detects image inline attachment by cid reference when Content-Disposition is missing

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

nooty committed:
fix(mailbox): detect inline image attachments by cid when Content-Disposition is missing

nooty committed:
test(mailbox): add test for inline image detection via cid without Content-Disposition

nooty committed:
chore(mailbox): clean up inline image CID test

pranavrajs commented:
@nooty Thanks for the PR and the specs. Appreciate it! I did a quick skim of the code and it looks good overall. I’ll do a more thorough round of testing later this week or early next, and plan to get it out after that.

There are a few CI issues showing up. if you could take a look at those, that’d be helpful.

<img width="1259" height="320" alt="Screenshot 2025-07-15 at 7 49 31 PM" src="https://github.com/user-attachments/assets/677b1f53-fbac-4449-b2dd-11980b888ab1" />

nooty committed:
chore(mailbox): fix trailing whitespace to comply with RuboCop

nooty commented:
Hi, @pranavrajs 

Thanks for the feedback!

I noticed that one remaining RuboCop complaint is related to Metrics/ModuleLength on MailboxHelper, due to the module exceeding the 100-line limit. To resolve this without impacting the readability of the current code, I can extract the inline attachments processing logic into a new class (e.g., InlineAttachmentProcessor or similar).

Would you be open to this approach? If so, I’ll go ahead and make the necessary changes.

pranavrajs committed:
Merge branch 'develop' into fix/11948-inline-attachment-outlook

pranavrajs commented:
@nooty At the moment, let us not do that. Let me check the PR once before we make any further changes. Thanks.

sojan-official committed:
Merge branch 'develop' into fix/11948-inline-attachment-outlook

sony-mathew committed:
Merge branch 'develop' into fix/11948-inline-attachment-outlook

sony-mathew committed:
Merge branch 'develop' into fix/11948-inline-attachment-outlook

sony-mathew committed:
Merge branch 'develop' into fix/11948-inline-attachment-outlook

sony-mathew committed:
fix(mailbox): support encoded inline CID images

Extract inline attachment CID handling into a dedicated mailbox helper so MailboxHelper stays within RuboCop limits while preserving the existing comments.

Recognize and rewrite both raw cid: references and RFC 2392 URL-encoded Content-ID variants, including lowercase percent-encoded forms, and add specs for encoded CID detection and replacement.

sony-mathew committed:
Merge branch 'develop' into fix/11948-inline-attachment-outlook

sony-mathew reviewed (approved):
LGTM

sony-mathew commented:
Thank you @nooty  for the contribution!
